### ISSUE TITLE
Problem with daily plan file.

### ISSUE THREAD

**Issue by prvnpandey** [matsim-org/matsim-code-examples#274]
I have prepared a synthetic population from travel dairy provided by municipality, but my problem is some of the person plans are more than one day in this situation my output event file is created for 120 hours, which is more than one day, this is happening because activity is having time 00:00:00  so because of that it is running for more than one day. I tried to delete all these rows from the data but still, the problem persists is there any way to make plan file strictly for one day.
@kainagel 

I am uploading a plan file for reference.
[plan.zip](https://github.com/matsim-org/matsim-code-examples/files/4104779/plan.zip)

**Comment by kainagel** [comment]
Hello, 

from skimming through your plans file I could not find a plan that had an endTime later than 24:00:00.  Maybe point me to an affected agentId.  If your mobsim runs much longer, it might rather be a problem with your capacities, e.g. the network too "narrow" for the demand, or the demand attached to the network through low capacity links.  Have you looked at the simulation result with a visualizer, i.e. VIA or otfvis?

In general, the mobsim should work fine with plans that go beyond 24:00:00.  A problem will be the scoring, which assumes "24 hours wrap around".  The scoring can be replaced, but only in Java code.

Best wishes, Kai

**Comment by prvnpandey** [comment]
yes i have checked event file in via, my point is the eventfile which was created at the end of iteraton. the end time at which last event was written is (time = 6963000) which is approx 1900 hours. To make strict for one day in qsim i have edited endtime = 24:00:00, it is fine to define qsim endtime and second thing is what will happen with the activites after this time.

Thanks and Regards

**Comment by kainagel** [comment]
Two things:

(1) You need to find out what was wrong with that agent/those agents that needed 1900 hours.   I have often seen that they are not plausible; a typical reason is that very many persons are attached to a link with very low capacity, and it takes them a long time to get out.  Such things need to be corrected, often by fuzzifying the coordinates.  I have also seen that such things may simply happen because the area where activities take place is larger than the area where the network is available.  In this case, the network needs to cover a larger area.  Etc.

(2) Once you are sure that these are "legit" (i.e. _not_ the consequence of something bad), you can indeed stop the mobsim early.  24:00:00 is too early; 36:00:00 is normally ok.  A consequence will be that all agents that have not finished their plan by then will receive a very bad score.  These agents will influence your averages, so you need to make sure to remove them before you analyse results.  They also have trouble learning, i.e. they may not use a mode which is best for them.  Which is why it is in the end easier to identify the root causes of such behavior, eliminate it, and thus make sure that all agents can finish their plans before the mobsim ends.
